### ISSUE TITLE
AlphaHPP without 3D Secure (`3DSecureBypass` = `Y`) and with `Currency` = `AUD` and with a trailing zero in the payment amount: «Provider unable to process request» / «"ProviderName": "westpac"»

### ISSUE THREAD

**Issue by dmitrii-fediuk** [mage2pro/alphacommercehub#15]
![02](https://user-images.githubusercontent.com/83204/32761810-754a66d0-c906-11e7-89db-85a73d638b0e.png)

![01](https://user-images.githubusercontent.com/83204/32761816-7d32a18c-c906-11e7-9a39-aa02cb4b2ffd.png)

### General
```
Request URL: https://hubuat.alphacommercehub.com.au/pp/ac580978-81f0-4716-bfc3-3bc46eab8603
Request Method: POST
Status Code: 200 OK
Remote Address: 76.12.61.237:443
Referrer Policy: no-referrer-when-downgrade
```

### Response headers
```
HTTP/1.1 200 OK
Date: Tue, 14 Nov 2017 03:33:23 GMT
Server: Microsoft-IIS/8.5
Cache-Control: private
Content-Type: text/html; charset=utf-8
X-AspNetMvc-Version: 4.0
X-AspNet-Version: 4.0.30319
X-Powered-By: ASP.NET
Vary: Accept-Encoding
Content-Encoding: gzip
Content-Length: 7172
Keep-Alive: timeout=5, max=100
Connection: Keep-Alive
```

### Request headers
```
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9,ru;q=0.8
Cache-Control: no-cache
Connection: keep-alive
Content-Length: 1126
Content-Type: application/x-www-form-urlencoded
Cookie: __RequestVerificationToken=n-B5epNpZExPBpdLRWY_IOg_zxlnxkKjL1utioq1JrxE6rgCQN9BJ480lFpp8mWnGdzLeRMhjjnUk-PHF_BJBywBGWw7qNJYr6zSdMzZPcjvYobCiCRY5Eyc2vjI9cd0Pdv0_Q2; _timeZoneOffset=180; _ga=GA1.3.685466192.1508711293; ASP.NET_SessionId=gam2fam11wnk5cryudsqbxps; .ASPXAUTH=3ED353E66D8A6BF8D78C2CC3EE304A9AC0F02B9391057D31ACE1A74CE33F5D7FC2872FF60000802C6BF713AEC8EC7835F816BA5120B2A0521E45C5521D28A5AF78B5C838794351B5DE91E4A8E64BE8344535C633642DA75A90724ED9283621CAA46327BA202B1FB169ED905B653E1DF5F7B9A49DFEA2D150D65D6A87EE13DE1CF0BDF18F77232DF9ADFC3B77652260AD5F77B8A87DF3EC066A0A6A84A96CBDCC7CAB81C410D5E683EA06EEBE0B462A5C8D93DFD0; email=seb%40anterior.com.au
Host: hubuat.alphacommercehub.com.au
Origin: https://localhost.com:900
Pragma: no-cache
Referer: https://localhost.com:900/store/checkout/
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.89 Safari/537.36
```

### Form data
```
3DSecureBypass: Y
Amount: 160710
CancelURL: https://mage2.pro/sandbox/dfe-alphacommercehub/customerReturn
Capture: Y
City: Sydney
Country: AU
Currency: AUD
CustomerID: dfediuk@gmail.com
EmailAddress: dfediuk@gmail.com
FraudCheckBypass: N
IPAddress: 92.243.166.8
Language: en
MerchantDescriptor: Mage2PRO
MerchantID: ANTERIOR001
MerchantTxnID: 1104L733
Method: CC
OrderDetails[0][ItemAmount]: 45550
OrderDetails[0][ItemName]: Alligator Briefcase
OrderDetails[0][ItemQuantity]: 2
OrderDetails[0][ItemTaxExempt]: TRUE
OrderDetails[1][ItemAmount]: 56940
OrderDetails[1][ItemName]: New Very Prive 41 Black
OrderDetails[1][ItemQuantity]: 1
OrderDetails[1][ItemTaxExempt]: TRUE
OrderDetails[2][ItemAmount]: 12650
OrderDetails[2][ItemName]: Shipping
OrderDetails[2][ItemQuantity]: 1
OrderDetails[2][ItemTaxExempt]: TRUE
SocialID: 74041803683
State: Mage2PRO
Street1: 72 Liverpool St
SuccessURL: https://mage2.pro/sandbox/dfe-alphacommercehub/customerReturn
TelNo: 61282688888
TxnDetails: Alligator Briefcase 2 New Very Prive
UserId: 123
Zip: 2000
```

### General
```
Request URL: https://hubuat.alphacommercehub.com.au/pp/auth
Request Method: POST
Status Code: 200 OK
Remote Address: 76.12.61.237:443
Referrer Policy: no-referrer-when-downgrade
```

### Response headers
```
HTTP/1.1 200 OK
Date: Tue, 14 Nov 2017 03:34:36 GMT
Server: Microsoft-IIS/8.5
Cache-Control: private
Content-Type: application/json; charset=utf-8
X-AspNetMvc-Version: 4.0
X-AspNet-Version: 4.0.30319
X-Powered-By: ASP.NET
Content-Length: 1517
Keep-Alive: timeout=5, max=100
Connection: Keep-Alive
```

### Request headers
```
Accept:*/*
Accept-Encoding:gzip, deflate, br
Accept-Language:en-US,en;q=0.9,ru;q=0.8
Cache-Control: no-cache
Connection: keep-alive
Content-Length: 3021
Content-Type: application/x-www-form-urlencoded; charset=UTF-8
Cookie: __RequestVerificationToken=n-B5epNpZExPBpdLRWY_IOg_zxlnxkKjL1utioq1JrxE6rgCQN9BJ480lFpp8mWnGdzLeRMhjjnUk-PHF_BJBywBGWw7qNJYr6zSdMzZPcjvYobCiCRY5Eyc2vjI9cd0Pdv0_Q2; _timeZoneOffset=180; _ga=GA1.3.685466192.1508711293; ASP.NET_SessionId=gam2fam11wnk5cryudsqbxps; .ASPXAUTH=3ED353E66D8A6BF8D78C2CC3EE304A9AC0F02B9391057D31ACE1A74CE33F5D7FC2872FF60000802C6BF713AEC8EC7835F816BA5120B2A0521E45C5521D28A5AF78B5C838794351B5DE91E4A8E64BE8344535C633642DA75A90724ED9283621CAA46327BA202B1FB169ED905B653E1DF5F7B9A49DFEA2D150D65D6A87EE13DE1CF0BDF18F77232DF9ADFC3B77652260AD5F77B8A87DF3EC066A0A6A84A96CBDCC7CAB81C410D5E683EA06EEBE0B462A5C8D93DFD0; email=seb%40anterior.com.au
Host: hubuat.alphacommercehub.com.au
Origin: https://hubuat.alphacommercehub.com.au
Pragma: no-cache
Referer: https://hubuat.alphacommercehub.com.au/pp/ac580978-81f0-4716-bfc3-3bc46eab8603
User-Agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.89 Safari/537.36
X-APC-MerchantTxnID: 1104L733
X-APC-PPSessionId: d2867dce-2089-4b7a-bcaf-5dc0ce747bd3
X-Requested-With: XMLHttpRequest
```

### Form data
```
cardHolder: DMITRY FEDYUK
cardNumber: 5313 5700 0000 0011
cardExpiryDate: 07 / 21
cardCVV: 222
country: AU
street1: 72 Liverpool St
street2:
city: Sydney
zip: 2000
emailAddress:
token: eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJwYXlsb2FkIjoie1wiVXNlcklkXCI6XCIxMjNcIixcIk1lcmNoYW50SURcIjpcIkFOVEVSSU9SMDAxXCIsXCJNZXJjaGFudFR4bklEXCI6XCIxMTA0TDczM1wiLFwiQ3VycmVuY3lcIjpcIkFVRFwiLFwiQW1vdW50XCI6XCIxNjA3MTBcIixcIlRyYW5zYWN0aW9uVGltZXN0YW1wXCI6bnVsbCxcIk1ldGhvZFwiOlwiQ0NcIixcIkxhbmd1YWdlXCI6XCJlblwiLFwiVHhuRGV0YWlsc1wiOlwiQWxsaWdhdG9yIEJyaWVmY2FzZSAyIE5ldyBWZXJ5IFByaXZlXCIsXCJDYXB0dXJlXCI6XCJZXCIsXCJFbWFpbEFkZHJlc3NcIjpcImRmZWRpdWtAZ21haWwuY29tXCIsXCJJUEFkZHJlc3NcIjpcIjkyLjI0My4xNjYuOFwiLFwiVGVsTm9cIjpcIjYxMjgyNjg4ODg4XCIsXCJDdXN0b21lcklEXCI6XCJkZmVkaXVrQGdtYWlsLmNvbVwiLFwiU29jaWFsSURcIjpcIjc0MDQxODAzNjgzXCIsXCJTdHJlZXQxXCI6XCI3MiBMaXZlcnBvb2wgU3RcIixcIlN0cmVldDJcIjpudWxsLFwiQ2l0eVwiOlwiU3lkbmV5XCIsXCJaaXBcIjpcIjIwMDBcIixcIlN0YXRlXCI6XCJNYWdlMlBST1wiLFwiQ291bnRyeVwiOlwiQVVcIixcIk1lcmNoYW50RGVzY3JpcHRvclwiOlwiTWFnZTJQUk9cIixcIlN1Y2Nlc3NVUkxcIjpcImh0dHBzOi8vbWFnZTIucHJvL3NhbmRib3gvZGZlLWFscGhhY29tbWVyY2VodWIvY3VzdG9tZXJSZXR1cm5cIixcIlRocmVlRFNlY3VyZUJ5cGFzc1wiOlwiWVwiLFwiQ2FyZFR5cGVcIjpcImFtZXgsamNiLG1hc3RlcmNhcmQsZGlzY292ZXIsbWFzdGVyY2FyZGRlYml0LHZpc2EsdmlzYWRlYml0LHVuaW9ucGF5LG1hZXN0cm8sdWF0cCxkaW5lcnNjbHViXCIsXCJGcmF1ZENoZWNrQnlwYXNzXCI6XCJOXCIsXCJJbnN0YWxsbWVudFwiOm51bGwsXCJQYXlMYXRlck1vbnRoc1wiOm51bGwsXCJQYXlOdW1iZXJNb250aHNcIjpudWxsLFwiSW5zdGFsbG1lbnRJRFwiOm51bGwsXCJJbnN0YWxsbWVudFR5cGVcIjpudWxsLFwiQ2FuY2VsVVJMXCI6XCJodHRwczovL21hZ2UyLnByby9zYW5kYm94L2RmZS1hbHBoYWNvbW1lcmNlaHViL2N1c3RvbWVyUmV0dXJuXCIsXCJDYXJkTnVtYmVyXCI6bnVsbCxcIkNhcmRIb2xkZXJcIjpudWxsLFwiQ2FyZEV4cGlyeU1vbnRoXCI6bnVsbCxcIkNhcmRFeHBpcnlZZWFyXCI6bnVsbCxcIkNhcmRDVlZcIjpudWxsLFwiQ2hhbm5lbFR5cGVcIjpcIjA3XCIsXCJTZWN1cmVUb2tlblwiOm51bGwsXCJTZXNzaW9uSWRcIjpudWxsLFwiVGF4QWRkZWRcIjpmYWxzZSxcIlRheFJhdGVcIjowLjAsXCJOb3RpZmljYXRpb25FbWFpbFwiOm51bGwsXCJSZXR1cm5UZXh0XCI6bnVsbCxcIlJlcGx5VXJsXCI6bnVsbCxcIlVwZGF0ZVVSTFwiOm51bGwsXCJPcmlnaW5cIjpcIkhQUFwiLFwiVHJhbnNhY3Rpb25UeXBlXCI6bnVsbCxcIkN1c3RvbTFcIjpudWxsLFwiUGF5V2l0aFBvaW50c1wiOm51bGwsXCJNZXJjaGFudERhdGFcIjpudWxsLFwiVGVybWluYWxcIjpudWxsLFwiUGF5bWVudFByb2Nlc3NvckFQSUlkXCI6bnVsbCxcIk9yZGVyRGV0YWlsc1wiOlt7XCJJdGVtTmFtZVwiOm51bGwsXCJJdGVtUXVhbnRpdHlcIjowLFwiSXRlbUFtb3VudFwiOm51bGwsXCJJdGVtUHJpY2VcIjpudWxsLFwiSXRlbVRheEV4ZW1wdFwiOmZhbHNlfSx7XCJJdGVtTmFtZVwiOm51bGwsXCJJdGVtUXVhbnRpdHlcIjowLFwiSXRlbUFtb3VudFwiOm51bGwsXCJJdGVtUHJpY2VcIjpudWxsLFwiSXRlbVRheEV4ZW1wdFwiOmZhbHNlfSx7XCJJdGVtTmFtZVwiOm51bGwsXCJJdGVtUXVhbnRpdHlcIjowLFwiSXRlbUFtb3VudFwiOm51bGwsXCJJdGVtUHJpY2VcIjpudWxsLFwiSXRlbVRheEV4ZW1wdFwiOmZhbHNlfV0sXCJDdXN0b21GaWVsZHNcIjpbXSxcIkFjY2VwdDNEU0F1dGhlbnRpY2F0aW9uVW5hdmFpbGFibGVcIjpmYWxzZSxcIkFjY2VwdDNEU0Vucm9sQnlwYXNzXCI6ZmFsc2UsXCJQUFNlc3Npb25JZFwiOlwiZWE2YjRmNjItMDY4Yy00YzU2LWIyMDYtMWMwZjRjY2YwMWFlXCJ9In0.WfIyXtEk1f0Fnc9vCUZnkZPZeS8y9USTpPi9gzG92gk
cardExpiryMonth: 7
cardExpiryYear: 2021
payPageID: ac580978-81f0-4716-bfc3-3bc46eab8603
merchantSequenceNumber:
clientIP: 92.243.166.8
```

### Response
```
{
	"Result": {
		"ResponseCode": "1076",
		"ResponseMessage": "Provider unable to process request",
		"ResponseTimestamp": "14112017033437",
		"MerchantID": "ANTERIOR001",
		"MerchantTxnID": "1104L733",
		"PaymentID": "107063741",
		"Currency": "AUD",
		"Amount": "160710",
		"BillCurrency": null,
		"BillAmount": null,
		"CardHolder": "DMITRY FEDYUK",
		"CardType": "MASTERCARD",
		"Custom1": null,
		"Custom2": null,
		"Custom3": null,
		"Custom4": null,
		"Custom5": null
	},
	"ProviderResps": [
		{
			"ProviderName": "westpac",
			"ProviderOrder": "1",
			"ProviderResponse": "71",
			"ProviderReference": "7001757683",
			"ResponseCode": "1076",
			"ResponseTimestamp": "14112017143437"
		}
	],
	"Country": {
		"CountryName": null,
		"CountryCode": null
	},
	"MethodResult": {
		"AuthCode": null,
		"CVVResult": "M",
		"AVSStreetResult": null,
		"AVSZipResult": null,
		"SecureToken": null,
		"Status": "1",
		"AmountReceived": null,
		"RedirectUrl": null,
		"FeeAmount": null,
		"Custom1": null,
		"Custom2": null,
		"Custom3": null,
		"Custom4": null,
		"Custom5": null,
		"ACSUrl": null,
		"PAReq": null
	},
	"ThreeDSecure": {
		"AuthenticationValue": null,
		"AuthenticationStatus": null,
		"Enrolled": null,
		"XID": null,
		"ECI": null,
		"PARes": null,
		"CAVVAlgorithm": null,
		"Custom1": null,
		"Custom2": null,
		"Custom3": null,
		"Custom4": null,
		"Custom5": null
	},
	"TokenResult": {
		"Custom1": null,
		"Custom2": null,
		"Custom3": null,
		"Custom4": null,
		"Custom5": null
	},
	"PaymentInfo": {
		"PaymentBrand": "MASTERCARD",
		"PaymentBrandType": "CREDIT",
		"PaymentBrandCategory": "CORPORATE",
		"PaymentIssuer": null,
		"PaymentIssuerCountry": "AU",
		"BIN": "531357"
	},
	"FraudResult": {
		"RiskCategory": null,
		"Score": null,
		"Status": null
	},
	"FraudChecks": [],
	"GeoChecks": [],
	"LoyaltyResult": null
}
```

**Comment by dmitrii-fediuk** [comment]
I suspected that the failure is caused by a [mismatch between the `Amount` value and the aggregated order items amounts](https://github.com/mage2pro/alphacommercehub/issues/16), but making the amounts the same does not help:

![03](https://user-images.githubusercontent.com/83204/32763016-118cfa24-c90e-11e7-83b9-fb094347eb7e.png)

![01](https://user-images.githubusercontent.com/83204/32763021-15d4858e-c90e-11e7-9c73-e2051be60102.png)

## 1. The initial redirection to AlphaHPP
### General
```
Request URL: https://hubuat.alphacommercehub.com.au/pp/ac580978-81f0-4716-bfc3-3bc46eab8603
Request Method: POST
Status Code: 200 OK
Remote Address: 76.12.61.237:443
Referrer Policy: no-referrer-when-downgrade
```

### Response headers
```
HTTP/1.1 200 OK
Date: Tue, 14 Nov 2017 04:24:14 GMT
Server: Microsoft-IIS/8.5
Cache-Control: private
Content-Type: text/html; charset=utf-8
X-AspNetMvc-Version: 4.0
X-AspNet-Version: 4.0.30319
X-Powered-By: ASP.NET
Vary: Accept-Encoding
Content-Encoding: gzip
Content-Length: 7177
Keep-Alive: timeout=5, max=100
Connection: Keep-Alive
```

### Request headers
```
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9,ru;q=0.8
Cache-Control: no-cache
Connection: keep-alive
Content-Length: 1294
Content-Type: application/x-www-form-urlencoded
Cookie: __RequestVerificationToken=n-B5epNpZExPBpdLRWY_IOg_zxlnxkKjL1utioq1JrxE6rgCQN9BJ480lFpp8mWnGdzLeRMhjjnUk-PHF_BJBywBGWw7qNJYr6zSdMzZPcjvYobCiCRY5Eyc2vjI9cd0Pdv0_Q2; _timeZoneOffset=180; _ga=GA1.3.685466192.1508711293; ASP.NET_SessionId=gam2fam11wnk5cryudsqbxps; .ASPXAUTH=3ED353E66D8A6BF8D78C2CC3EE304A9AC0F02B9391057D31ACE1A74CE33F5D7FC2872FF60000802C6BF713AEC8EC7835F816BA5120B2A0521E45C5521D28A5AF78B5C838794351B5DE91E4A8E64BE8344535C633642DA75A90724ED9283621CAA46327BA202B1FB169ED905B653E1DF5F7B9A49DFEA2D150D65D6A87EE13DE1CF0BDF18F77232DF9ADFC3B77652260AD5F77B8A87DF3EC066A0A6A84A96CBDCC7CAB81C410D5E683EA06EEBE0B462A5C8D93DFD0; email=seb%40anterior.com.au
Host: hubuat.alphacommercehub.com.au
Origin: https://localhost.com:900
Pragma: no-cache
Referer: https://localhost.com:900/store/checkout/
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.89 Safari/537.36
```

### Form data
```
3DSecureBypass: Y
Amount: 160710
CancelURL: https://mage2.pro/sandbox/dfe-alphacommercehub/customerReturn
Capture: Y
City: Sydney
Country: AU
Currency: AUD
CustomerID: dfediuk@gmail.com
EmailAddress: dfediuk@gmail.com
FraudCheckBypass: N
IPAddress: 92.243.166.8
Language: en
MerchantDescriptor: Mage2PRO
MerchantID: ANTERIOR001
MerchantTxnID: 1104L735
Method: CC
OrderDetails[0][ItemAmount]: 45550
OrderDetails[0][ItemName]: Alligator Briefcase
OrderDetails[0][ItemQuantity]: 2
OrderDetails[0][ItemTaxExempt]: TRUE
OrderDetails[1][ItemAmount]: 56940
OrderDetails[1][ItemName]: New Very Prive 41 Black
OrderDetails[1][ItemQuantity]: 1
OrderDetails[1][ItemTaxExempt]: TRUE
OrderDetails[2][ItemAmount]: 12650
OrderDetails[2][ItemName]: Shipping
OrderDetails[2][ItemQuantity]: 1
OrderDetails[2][ItemTaxExempt]: TRUE
OrderDetails[3][ItemAmount]: 20
OrderDetails[3][ItemName]: Adjustment
OrderDetails[3][ItemQuantity]: 1
OrderDetails[3][ItemTaxExempt]: TRUE
SocialID: 74041803683
State: Mage2PRO
Street1: 72 Liverpool St
SuccessURL: https://mage2.pro/sandbox/dfe-alphacommercehub/customerReturn
TelNo: 61282688888
TxnDetails: Alligator Briefcase 2 New Very Prive
UserId: 123
Zip: 2000
```

## 2. The `https://hubuat.alphacommercehub.com.au/pp/auth` request
### General
```
Request URL: https://hubuat.alphacommercehub.com.au/pp/auth
Request Method: POST
Status Code: 200 OK
Remote Address: 76.12.61.237:443
Referrer Policy: no-referrer-when-downgrade
```

### Response headers
```
HTTP/1.1 200 OK
Date: Tue, 14 Nov 2017 04:24:54 GMT
Server: Microsoft-IIS/8.5
Cache-Control: private
Content-Type: application/json; charset=utf-8
X-AspNetMvc-Version: 4.0
X-AspNet-Version: 4.0.30319
X-Powered-By: ASP.NET
Content-Length: 1517
Keep-Alive: timeout=5, max=100
Connection: Keep-Alive
```

### Request headers
```
Accept: */*
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9,ru;q=0.8
Cache-Control: no-cache
Connection: keep-alive
Content-Length: 3157
Content-Type: application/x-www-form-urlencoded; charset=UTF-8
Cookie: __RequestVerificationToken=n-B5epNpZExPBpdLRWY_IOg_zxlnxkKjL1utioq1JrxE6rgCQN9BJ480lFpp8mWnGdzLeRMhjjnUk-PHF_BJBywBGWw7qNJYr6zSdMzZPcjvYobCiCRY5Eyc2vjI9cd0Pdv0_Q2; _timeZoneOffset=180; _ga=GA1.3.685466192.1508711293; ASP.NET_SessionId=gam2fam11wnk5cryudsqbxps; .ASPXAUTH=3ED353E66D8A6BF8D78C2CC3EE304A9AC0F02B9391057D31ACE1A74CE33F5D7FC2872FF60000802C6BF713AEC8EC7835F816BA5120B2A0521E45C5521D28A5AF78B5C838794351B5DE91E4A8E64BE8344535C633642DA75A90724ED9283621CAA46327BA202B1FB169ED905B653E1DF5F7B9A49DFEA2D150D65D6A87EE13DE1CF0BDF18F77232DF9ADFC3B77652260AD5F77B8A87DF3EC066A0A6A84A96CBDCC7CAB81C410D5E683EA06EEBE0B462A5C8D93DFD0; email=seb%40anterior.com.au
Host: hubuat.alphacommercehub.com.au
Origin: https://hubuat.alphacommercehub.com.au
Pragma: no-cache
Referer: https://hubuat.alphacommercehub.com.au/pp/ac580978-81f0-4716-bfc3-3bc46eab8603
User-Agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.89 Safari/537.36
X-APC-MerchantTxnID: 1104L735
X-APC-PPSessionId: 94411908-0341-4bb8-a3f1-d6b1ea5bbb10
X-Requested-With: XMLHttpRequest
```

### Form data
```
cardHolder: DMITRY FEDYUK
cardNumber: 5313 5700 0000 0011
cardExpiryDate: 07 / 21
cardCVV: 222
country: AU
street1: 72 Liverpool St
street2:
city: Sydney
zip: 2000
emailAddress:
token: eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJwYXlsb2FkIjoie1wiVXNlcklkXCI6XCIxMjNcIixcIk1lcmNoYW50SURcIjpcIkFOVEVSSU9SMDAxXCIsXCJNZXJjaGFudFR4bklEXCI6XCIxMTA0TDczNVwiLFwiQ3VycmVuY3lcIjpcIkFVRFwiLFwiQW1vdW50XCI6XCIxNjA3MTBcIixcIlRyYW5zYWN0aW9uVGltZXN0YW1wXCI6bnVsbCxcIk1ldGhvZFwiOlwiQ0NcIixcIkxhbmd1YWdlXCI6XCJlblwiLFwiVHhuRGV0YWlsc1wiOlwiQWxsaWdhdG9yIEJyaWVmY2FzZSAyIE5ldyBWZXJ5IFByaXZlXCIsXCJDYXB0dXJlXCI6XCJZXCIsXCJFbWFpbEFkZHJlc3NcIjpcImRmZWRpdWtAZ21haWwuY29tXCIsXCJJUEFkZHJlc3NcIjpcIjkyLjI0My4xNjYuOFwiLFwiVGVsTm9cIjpcIjYxMjgyNjg4ODg4XCIsXCJDdXN0b21lcklEXCI6XCJkZmVkaXVrQGdtYWlsLmNvbVwiLFwiU29jaWFsSURcIjpcIjc0MDQxODAzNjgzXCIsXCJTdHJlZXQxXCI6XCI3MiBMaXZlcnBvb2wgU3RcIixcIlN0cmVldDJcIjpudWxsLFwiQ2l0eVwiOlwiU3lkbmV5XCIsXCJaaXBcIjpcIjIwMDBcIixcIlN0YXRlXCI6XCJNYWdlMlBST1wiLFwiQ291bnRyeVwiOlwiQVVcIixcIk1lcmNoYW50RGVzY3JpcHRvclwiOlwiTWFnZTJQUk9cIixcIlN1Y2Nlc3NVUkxcIjpcImh0dHBzOi8vbWFnZTIucHJvL3NhbmRib3gvZGZlLWFscGhhY29tbWVyY2VodWIvY3VzdG9tZXJSZXR1cm5cIixcIlRocmVlRFNlY3VyZUJ5cGFzc1wiOlwiWVwiLFwiQ2FyZFR5cGVcIjpcImFtZXgsamNiLG1hc3RlcmNhcmQsZGlzY292ZXIsbWFzdGVyY2FyZGRlYml0LHZpc2EsdmlzYWRlYml0LHVuaW9ucGF5LG1hZXN0cm8sdWF0cCxkaW5lcnNjbHViXCIsXCJGcmF1ZENoZWNrQnlwYXNzXCI6XCJOXCIsXCJJbnN0YWxsbWVudFwiOm51bGwsXCJQYXlMYXRlck1vbnRoc1wiOm51bGwsXCJQYXlOdW1iZXJNb250aHNcIjpudWxsLFwiSW5zdGFsbG1lbnRJRFwiOm51bGwsXCJJbnN0YWxsbWVudFR5cGVcIjpudWxsLFwiQ2FuY2VsVVJMXCI6XCJodHRwczovL21hZ2UyLnByby9zYW5kYm94L2RmZS1hbHBoYWNvbW1lcmNlaHViL2N1c3RvbWVyUmV0dXJuXCIsXCJDYXJkTnVtYmVyXCI6bnVsbCxcIkNhcmRIb2xkZXJcIjpudWxsLFwiQ2FyZEV4cGlyeU1vbnRoXCI6bnVsbCxcIkNhcmRFeHBpcnlZZWFyXCI6bnVsbCxcIkNhcmRDVlZcIjpudWxsLFwiQ2hhbm5lbFR5cGVcIjpcIjA3XCIsXCJTZWN1cmVUb2tlblwiOm51bGwsXCJTZXNzaW9uSWRcIjpudWxsLFwiVGF4QWRkZWRcIjpmYWxzZSxcIlRheFJhdGVcIjowLjAsXCJOb3RpZmljYXRpb25FbWFpbFwiOm51bGwsXCJSZXR1cm5UZXh0XCI6bnVsbCxcIlJlcGx5VXJsXCI6bnVsbCxcIlVwZGF0ZVVSTFwiOm51bGwsXCJPcmlnaW5cIjpcIkhQUFwiLFwiVHJhbnNhY3Rpb25UeXBlXCI6bnVsbCxcIkN1c3RvbTFcIjpudWxsLFwiUGF5V2l0aFBvaW50c1wiOm51bGwsXCJNZXJjaGFudERhdGFcIjpudWxsLFwiVGVybWluYWxcIjpudWxsLFwiUGF5bWVudFByb2Nlc3NvckFQSUlkXCI6bnVsbCxcIk9yZGVyRGV0YWlsc1wiOlt7XCJJdGVtTmFtZVwiOm51bGwsXCJJdGVtUXVhbnRpdHlcIjowLFwiSXRlbUFtb3VudFwiOm51bGwsXCJJdGVtUHJpY2VcIjpudWxsLFwiSXRlbVRheEV4ZW1wdFwiOmZhbHNlfSx7XCJJdGVtTmFtZVwiOm51bGwsXCJJdGVtUXVhbnRpdHlcIjowLFwiSXRlbUFtb3VudFwiOm51bGwsXCJJdGVtUHJpY2VcIjpudWxsLFwiSXRlbVRheEV4ZW1wdFwiOmZhbHNlfSx7XCJJdGVtTmFtZVwiOm51bGwsXCJJdGVtUXVhbnRpdHlcIjowLFwiSXRlbUFtb3VudFwiOm51bGwsXCJJdGVtUHJpY2VcIjpudWxsLFwiSXRlbVRheEV4ZW1wdFwiOmZhbHNlfSx7XCJJdGVtTmFtZVwiOm51bGwsXCJJdGVtUXVhbnRpdHlcIjowLFwiSXRlbUFtb3VudFwiOm51bGwsXCJJdGVtUHJpY2VcIjpudWxsLFwiSXRlbVRheEV4ZW1wdFwiOmZhbHNlfV0sXCJDdXN0b21GaWVsZHNcIjpbXSxcIkFjY2VwdDNEU0F1dGhlbnRpY2F0aW9uVW5hdmFpbGFibGVcIjpmYWxzZSxcIkFjY2VwdDNEU0Vucm9sQnlwYXNzXCI6ZmFsc2UsXCJQUFNlc3Npb25JZFwiOlwiMTFmYzgyNWQtNjRmOC00ZmQzLWFiMjEtNjNhMDQ2OTM3MDk0XCJ9In0.TdFosQOlSKZU__vl61z_fPDuhxzNe0OXeH5bOLw-Lh4
cardExpiryMonth: 7
cardExpiryYear: 2021
payPageID: ac580978-81f0-4716-bfc3-3bc46eab8603
merchantSequenceNumber:
clientIP: 92.243.166.8
```

### Response
```
{
	"Result": {
		"ResponseCode": "1076",
		"ResponseMessage": "Provider unable to process request",
		"ResponseTimestamp": "14112017042455",
		"MerchantID": "ANTERIOR001",
		"MerchantTxnID": "1104L735",
		"PaymentID": "107065879",
		"Currency": "AUD",
		"Amount": "160710",
		"BillCurrency": null,
		"BillAmount": null,
		"CardHolder": "DMITRY FEDYUK",
		"CardType": "MASTERCARD",
		"Custom1": null,
		"Custom2": null,
		"Custom3": null,
		"Custom4": null,
		"Custom5": null
	},
	"ProviderResps": [
		{
			"ProviderName": "westpac",
			"ProviderOrder": "1",
			"ProviderResponse": "71",
			"ProviderReference": "7001757915",
			"ResponseCode": "1076",
			"ResponseTimestamp": "14112017152455"
		}
	],
	"Country": {
		"CountryName": null,
		"CountryCode": null
	},
	"MethodResult": {
		"AuthCode": null,
		"CVVResult": "M",
		"AVSStreetResult": null,
		"AVSZipResult": null,
		"SecureToken": null,
		"Status": "1",
		"AmountReceived": null,
		"RedirectUrl": null,
		"FeeAmount": null,
		"Custom1": null,
		"Custom2": null,
		"Custom3": null,
		"Custom4": null,
		"Custom5": null,
		"ACSUrl": null,
		"PAReq": null
	},
	"ThreeDSecure": {
		"AuthenticationValue": null,
		"AuthenticationStatus": null,
		"Enrolled": null,
		"XID": null,
		"ECI": null,
		"PARes": null,
		"CAVVAlgorithm": null,
		"Custom1": null,
		"Custom2": null,
		"Custom3": null,
		"Custom4": null,
		"Custom5": null
	},
	"TokenResult": {
		"Custom1": null,
		"Custom2": null,
		"Custom3": null,
		"Custom4": null,
		"Custom5": null
	},
	"PaymentInfo": {
		"PaymentBrand": "MASTERCARD",
		"PaymentBrandType": "CREDIT",
		"PaymentBrandCategory": "CORPORATE",
		"PaymentIssuer": null,
		"PaymentIssuerCountry": "AU",
		"BIN": "531357"
	},
	"FraudResult": {
		"RiskCategory": null,
		"Score": null,
		"Status": null
	},
	"FraudChecks": [],
	"GeoChecks": [],
	"LoyaltyResult": null
}
```

**Comment by dmitrii-fediuk** [comment]
It is fixed by #17 (The last 3 digits of every payment amount should be «000» in the test mode for Westpac).
